spec/surfnperf_spec.js

```diff
@@ -1096,5 +1096,136 @@ define('spec/surfnperf_spec', [
 
     });
 
+
+    describe('#chromeLoadTimes', function() {
+      it('returns a reference to window.chrome.loadTimes() if the browser supports it, otherwise undefined', function() {
+        if(window.chrome && window.chrome.loadTimes) {
+          expect(SurfNPerf.chromeLoadTimes()).toEqual(window.chrome.loadTimes());
+        } else {
+          expect(SurfNPerf.chromeLoadTimes()).toEqual(undefined);
+        }
+      });
+    });
+
+    describe('#msFirstPaint', function() {
+      it('returns a reference to window.performance.timing.msFirstPaint if the browser supports it, otherwise undefined', function() {
+        if(window.performance && window.performance.timing && window.performance.timing.msFirstPaint) {
+          expect(SurfNPerf.msFirstPaint()).toEqual(window.performance.timing.msFirstPaint);
+        } else {
+          expect(SurfNPerf.msFirstPaint()).toEqual(undefined);
+        }
+      });
+    });
+
+    describe('#getFirstPaint', function() {
+      describe('when the client is using Chrome', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn({
+            firstPaintTime: 1446480965.123456
+          });
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(SurfNPerf.duration('navigationStart', 'chromePaint'));
+        });
+      });
+
+      describe('when the client is using IE/Edge', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(1776480965.123457);
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
```

Comment:
you can remove this `SurfNPerf.performanceTiming` spy now that it's handled in the `beforeEach`